packages/blade/src/components/Carousel/__tests__/Carousel.web.test.tsx

```diff
@@ -0,0 +1,437 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
+/* eslint-disable @typescript-eslint/require-await */
+import { fireEvent, act } from '@testing-library/react';
+import { Carousel } from '../Carousel';
+import { CarouselItem } from '../CarouselItem';
+import { useCarouselContext } from '../CarouselContext';
+import renderWithTheme from '~utils/testing/renderWithTheme.web';
+
+const TestimonialCard = ({ index }: { index: number }): React.ReactElement => {
+  const { activeSlide } = useCarouselContext();
+  return (
+    <div>
+      <p>{index}</p>
+      {/* A hack to get the internal state of the carousel so that i can assert the tests */}
+      {/* Users won't be able to do this since useCarouselContext is not gonna be exported */}
+      <p data-testid="active-slide">{activeSlide}</p>
+    </div>
+  );
+};
+
+beforeAll(() => {
```

Comment:
Revert the window object to the original state in `afterAll`?

Reply:
Done.